File: tests/nodejs/cases/case_174_indirect_container_write_08.js
/**
 * Task case 174: indirect_container_write_08 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case174IndirectContainerWrite08(input) {
  const taskName = 'indirect_container_write_08';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const envelope = { wrapped: payload };
  // ESCAPE: payload stored via indirection into retained sink.
  retainedAudit.push(envelope.wrapped);
  return 'ok';
}

module.exports = {
  case174IndirectContainerWrite08,
};

File: tests/go/cases/case_174_indirect_container_write_08.go
package escape_tests

var retainedCase174 = []map[string]string{}

func Case174IndirectContainerWrite08(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "indirect_container_write_08",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "indirect_container_write_08:" + raw,
	}
	envelope := map[string]map[string]string{"wrapped": payload}
	// ESCAPE: indirection writes payload into retained sink.
	retainedCase174 = append(retainedCase174, envelope["wrapped"])
	return "ok"
}
